### PR TITLE
2048: Better scoring

### PR DESCRIPTION
**Pull request by alimpfard**
> The scoring system is a bit lame

No longer!
The added score is logarithmically proportional to the merged tiles, and reversely proportional to the game time (i.e. the more turns you take, the less score you get).

Also moves out the undo action to the app menu.

### CODE REVIEW

**Comment by awesomekling** [comment]
I just tried the original 2048 game and it gives you 8 points when combining 4+4 into an 8 tile, 16 points when combining 8+8 into a 16 tile, etc.

**Comment by alimpfard** [comment]
But then that wouldn't incentivise finishing the game in less turns...
I could do that, but I just thought that this is more interesting.

File: Games/2048/main.cpp [hunk 1]
```diff
@@ -75,6 +75,9 @@ int main(int argc, char** argv)
     app_menu.add_action(GUI::Action::create("New game", { Mod_None, Key_F2 }, [&](auto&) {
         game.reset();
     }));
+    app_menu.add_action(GUI::Action::create("Undo", { Mod_None, Key_U }, [&](auto&) {
+        game.undo();
+    }));
```
Comment (awesomekling) [inline Games/2048/main.cpp:80]: Let's use `GUI::CommonActions::make_undo_action` so we get the right icon and shortcut :)

Comment (alimpfard) [reply Games/2048/main.cpp:80]: Oh we have that? neat!

**Pull request merged**